### DISCUSSION TITLE
Enable only one branch for deployment in vercel.json?

### DISCUSSION THREAD

**Discussion by gabrielgrs** [vercel/vercel#10636] [Help]
### Question

Hello!

Is there any way to skip all deployments except for a specific branch?
For example, I have a preview branch and a main branch that I want only to trigger the deployment.
I don't want to keep mapping all the branches, for example, something like "feat/*" doesn't become deploy.

```json
{
  "git": {
    "deploymentEnabled": {
      "preview": true,
      "main": false,
      "development": false,
      "feat/*": false
    }
  }
}
```

**Comment by jciurc** [comment]
I don't think it's possible. You can try deleting the deployment rules in `vercel.json` and going to your project settings page in the git section at the very bottom and setting it to build only for production. Not sure how to customize this to add more branches, probably need to create a custom command. 

![image](https://github.com/vercel/vercel/assets/24639525/ad28d4d7-0e9f-4a23-a054-f3d7d6908f74)

Another solution could be to disable auto deployment for every branch in the `vercel.json` file with below, then deploying manually through the Vercel CLI. I personally haven't tried this, but it's worth exploring. 
```json
{
  "git": {
    "deploymentEnabled": false
  }
}
```
